src/test/java/org/opensearch/ad/transport/ADJobParameterActionTests.java

```diff
@@ -0,0 +1,72 @@
+package org.opensearch.ad.transport;
+
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.opensearch.ad.TestHelpers;
+import org.opensearch.ad.util.RestHandlerUtils;
+import org.opensearch.common.bytes.BytesReference;
+import org.opensearch.common.io.stream.BytesStreamOutput;
+import org.opensearch.common.io.stream.NamedWriteableAwareStreamInput;
+import org.opensearch.common.io.stream.NamedWriteableRegistry;
+import org.opensearch.common.xcontent.*;
+import org.opensearch.extensions.action.ExtensionActionRequest;
+import org.opensearch.extensions.action.ExtensionActionResponse;
+import org.opensearch.jobscheduler.model.ExtensionJobParameter;
+import org.opensearch.jobscheduler.spi.JobDocVersion;
+import org.opensearch.jobscheduler.spi.ScheduledJobParameter;
+import org.opensearch.jobscheduler.transport.ExtensionJobActionRequest;
+import org.opensearch.jobscheduler.transport.ExtensionJobActionResponse;
+import org.opensearch.jobscheduler.transport.JobParameterRequest;
+import org.opensearch.jobscheduler.transport.JobParameterResponse;
+import org.opensearch.test.OpenSearchSingleNodeTestCase;
+
+public class ADJobParameterActionTests extends OpenSearchSingleNodeTestCase {
+    @Before
+    public void setUp() throws Exception {
+        super.setUp();
+    }
+
+    @Override
+    protected NamedWriteableRegistry writableRegistry() {
+        return getInstanceFromNode(NamedWriteableRegistry.class);
+    }
+
+    @Test
+    public void testExtensionActionRequest() throws Exception {
+        BytesStreamOutput out = new BytesStreamOutput();
+        XContentBuilder content = TestHelpers.randomXContent();
+        JobDocVersion jobDocVersion = new JobDocVersion(1L, 1L, 1L);
+        XContentParser parser = XContentHelper
+            .createParser(xContentRegistry(), LoggingDeprecationHandler.INSTANCE, BytesReference.bytes(content), XContentType.JSON);
+        JobParameterRequest jobParamRequest = new JobParameterRequest("token", parser, "id", jobDocVersion);
+        ExtensionActionRequest request = new ExtensionJobActionRequest<>(
+            RestHandlerUtils.EXTENSION_JOB_PARAMETER_ACTION_NAME,
+            jobParamRequest
+        );
+
+        request.writeTo(out);
+        NamedWriteableAwareStreamInput input = new NamedWriteableAwareStreamInput(out.bytes().streamInput(), writableRegistry());
```

Comment:
You probably need to flush `out` here to make sure the write finishes.  (Same comment multiple times for all the `writeTo(out)` in this PR, just noting this one.)

Reply:
ACK